### ISSUE TITLE
Multiple issues with Typescript

### ISSUE THREAD

**Issue by SimpleCreations** [mmazzarolo/react-native-dialog#118]
## Environment

Any

## Platforms

All

## Versions

- react-native-dialog: 8.2.0
- react-native: 0.64.1
- react: 17.0.2

## Description

Here are several issues I encountered when using the library with TypeScript:

1. `Dialog.Description` uses the `Text` element, however the `style` prop is declared as

        style?: ViewStyle
    and probably should be

        style?: StyleProp<TextStyle>
    to allow text styles and nested arrays, e.g.
    
    ```typescript
    <Dialog.Description style={[styles.description, {color: "#666"}]}>
    ```
    
    Every other place where `ViewStyle` is used instead of `TextStyle` and/or without `StyleProp` should probably be updated as well.

2. Cannot use conditional rendering inside `Dialog.Container`, e.g.

    ```typescript
        <Dialog.Container visible={visible}>
            <Dialog.Title>Title</Dialog.Title>
            {description != null &&
                <Dialog.Description>{description}</Dialog.Description>}
        </Dialog.Container>
    ```
    
        TS2322: Type 'false | Element' is not assignable to type 'ReactElement<any, NamedExoticComponent<{}>> & ReactNode'.
           Type 'false' is not assignable to type 'ReactElement<any, NamedExoticComponent<{}>> & ReactNode'.

    This is rendered as expected with type checking disabled.

3. `label` prop from `Dialog.Button` should probably be updated to allow custom components in addition to strings to take advantage of libraries like `react-intl`, e.g.

    ```typescript
        <Dialog.Button
            label={
                <FormattedMessage
                    id="common.cancelButton"
                    defaultMessage="Cancel" />}
            onPress={dismiss} />
    ```
    
    Right now only `string` is allowed in both Typescript definitions and prop types. However, with type checking disabled it renders as expected.

**Comment by mmazzarolo** [comment]
Hi @SimpleCreations , thanks for reporting :)

I think fixing typings of **1** and **3** could be a good first issue if you/anyone wants to contribute.  

The current behaviour for **2**, though, I think it's correct. You shouldn't render a `false` in your JSX, especially in React-Native. 
So this:
```js
description != null && <Dialog.Description>{description}</Dialog.Description>
```

Should be written like this:
```js
description != null ? <Dialog.Description>{description}</Dialog.Description> : undefined
```

(Or any similar approach to render null/undefined instead of a boolean)

**Comment by SimpleCreations** [comment]
@mmazzarolo thank you for the reply!

About **2** — it seems like `undefined` isn't allowed either:

```
TS2322: Type 'undefined' is not assignable to type 'ReactElement<any, NamedExoticComponent<{}>> & ReactNode'.
```

Also, rendering `false` is supported everywhere else in React+Typescript outside of `react-native-dialog` and is mentioned in the official React docs:
https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator
If it was considered a bad practice, I believe it wouldn't be shown there as an example :)

**Comment by mmazzarolo** [comment]
Ooops, I'm dumb.  
Thought the conditional was inside a Text node, which is notorious for complaining [about almost anything you throw at it in React-Native](https://stackoverflow.com/questions/52368342/invariant-violation-text-strings-must-be-rendered-within-a-text-component).  
We can/should update 2 too 👍

**Comment by EverStarck** [comment]
I have the same problem with the #2. To avoid a error I used a fragment to render other component inside modal. but it it's moved randomly or idk, so, in a cancel button I used the following:
```
<Dialog.Button
   label={closeButton ? 'Close' : ''}
   nPress={closeButton ? handleClose : null!}
/>
```
I know, it's ugly but it works 🤷‍♂️

**Comment by mmazzarolo** [comment]
Thanks for sharing a workaround @EverStarck. 

Open for PRs fixing it in the repo btw 👍

**Comment by mmazzarolo** [comment]
Closed thanks to @SimpleCreations in #121 🙌
